src/systems/a320_systems/src/pneumatic.rs

```diff
@@ -1,27 +1,3041 @@
+        );
+        self.fan_air_valve_pid.next_control_output(
+            sensors
+                .precooler_outlet_temperature()
+                .get::<degree_celsius>(),
+            Some(context.delta()),
+        );
+
+        self.engine_starter_valve_is_open = sensors.engine_starter_valve_is_open();
+
+        self.is_engine_bleed_pushbutton_auto =
+            overhead_panel.engine_bleed_pb_is_auto(self.engine_number);
+        self.is_engine_fire_pushbutton_released = is_engine_fire_pushbutton_released;
+
+        self.is_apu_bleed_valve_open = apu_bleed_valve.is_open();
+        self.is_apu_bleed_on = overhead_panel.apu_bleed_is_on();
+
+        self.cross_bleed_valve_selector = overhead_panel.cross_bleed_mode();
+        self.cross_bleed_valve_is_open = cross_bleed_valve.is_open();
+
+        self.engine_bleed_fault_light_monitor.update(
+            context,
+            !sensors.pressure_regulating_valve_is_open(),
+            !sensors.engine_starter_valve_is_open(),
+            !apu_bleed_valve.is_open(),
+            overhead_panel.apu_bleed_is_on(),
+            overhead_panel.cross_bleed_mode() == CrossBleedValveSelectorMode::Shut,
+            !cross_bleed_valve.is_open(),
+            sensors.precooler_inlet_pressure(),
+            sensors.precooler_outlet_temperature(),
+        );
+
+        self.update_fault_lights(overhead_panel);
+    }
+
+    fn update_fault_lights(&self, overhead_panel: &mut A320PneumaticOverheadPanel) {
+        if let Some(signal) = self.engine_bleed_fault_light_monitor.signal() {
+            overhead_panel
+                .set_engine_bleed_has_fault(self.engine_number, signal.fault_light_should_be_on());
+        }
+    }
+
+    pub fn operation_mode(&self) -> BleedMonitoringComputerChannelOperationMode {
+        self.operation_mode
+    }
+
+    pub fn set_operation_mode(&mut self, mode: BleedMonitoringComputerChannelOperationMode) {
+        self.operation_mode = mode;
+    }
+
+    pub fn or_none_if_slave(&self) -> Option<&BleedMonitoringComputerChannel> {
+        match self.operation_mode() {
+            BleedMonitoringComputerChannelOperationMode::Master => Some(self),
+            BleedMonitoringComputerChannelOperationMode::Slave => None,
+        }
+    }
+}
+impl ControllerSignal<HighPressureValveSignal> for BleedMonitoringComputerChannel {
+    fn signal(&self) -> Option<HighPressureValveSignal> {
+        if self.pressure_regulating_valve_is_closed {
+            return Some(HighPressureValveSignal::new_closed());
+        }
+
+        if self.high_pressure_compressor_pressure < Pressure::new::<psi>(18.) {
+            return Some(HighPressureValveSignal::new_closed());
+        }
+
+        Some(HighPressureValveSignal::new(Ratio::new::<ratio>(
+            self.high_pressure_valve_pid.output(),
+        )))
+    }
+}
+impl ControllerSignal<PressureRegulatingValveSignal> for BleedMonitoringComputerChannel {
+    fn signal(&self) -> Option<PressureRegulatingValveSignal> {
```

Comment:
**thought**: The same idiomatic Rust comment regarding usage of `return` applies here. 
**suggestion**: Do consider creating functions that wrap longer expressions, such as:
```rust
self.is_apu_bleed_on
            && self.is_apu_bleed_valve_open
            && (self.engine_number == 1 || self.cross_bleed_valve_is_open)
```